stages/org.osbuild.selinux

```diff
@@ -4,36 +4,49 @@ import pathlib
 import sys
 
 import osbuild.api
-from osbuild.util import selinux
+from osbuild.util import parsing, selinux
 
 
-def main(tree, options):
+def main(args):
+    options = args["options"]
     file_contexts = options.get("file_contexts")
     exclude_paths = options.get("exclude_paths")
 
     if file_contexts:
-        file_contexts = os.path.join(f"{tree}", options["file_contexts"])
-        if exclude_paths:
-            exclude_paths = [os.path.join(tree, p.lstrip("/")) for p in exclude_paths]
-        selinux.setfiles(file_contexts, os.fspath(tree), "", exclude_paths=exclude_paths)
+        if "://" not in file_contexts:
+            if file_contexts.startswith("/"):
+                file_contexts = f"tree://{file_contexts}"
+            else:
+                file_contexts = f"tree:///{file_contexts}"
+        file_contexts = parsing.parse_location(file_contexts, args)
 
     labels = options.get("labels", {})
-    for path, label in labels.items():
-        fullpath = os.path.join(tree, path.lstrip("/"))
-        selinux.setfilecon(fullpath, label)
 
-    if options.get("force_autorelabel", False):
-        stamp = pathlib.Path(tree, ".autorelabel")
-        # Creating just empty /.autorelabel resets only the type of files.
-        # To ensure that the full context is reset, we write "-F" into the file.
-        # This mimics the behavior of `fixfiles -F boot`. The "-F" option is
-        # then passed to `selinux-autorelabel` script [0].
-        # Note that this is missing from the selinux(8) and selinux_config(5) man-pages
-        # [0] https://src.fedoraproject.org/rpms/policycoreutils/blob/rawhide/f/selinux-autorelabel#_54
-        stamp.write_text("-F", encoding="utf-8")
+    # Set labels on each target
+    for target in options.get("targets", ["tree:///"]):
+        root, target = parsing.parse_location_ext(target, args)
+        target = target.lstrip("/")
+
+        if file_contexts:
+            if exclude_paths:
+                exclude_paths = [os.path.join(root, target, p.lstrip("/")) for p in exclude_paths]
+            selinux.setfiles(file_contexts, os.fspath(root), target, exclude_paths=exclude_paths)
+
+        for path, label in labels.items():
+            fullpath = os.path.join(root, path.lstrip("/"))
```

Comment:
What's the point to have `targets` just for `file_context` and leaving `labels` with `tree` only?  IMO, `labels` should also support mounts

Reply:
> IMO, `labels` should also support mounts

I agree. That's why I said:

> I'm semi sort of thinking we should just support full location specifiers for labels too.

So this would be valid:

```
    - type: org.osbuild.selinux
      options:
        labels:
          mount://root/path/to/file: system_u:object_r:boot_t:s0
          mount://root/path/to/other/file: system_u:object_r:var_t:s0
      devices:
        disk: ...
      mounts:
        - name: root
          source: disk
          target: /
```

Reply:
Such approach looks nice, but makes handling of  `force_autorelabel`  more complicated . Are you sure it should be used like that ?

Reply:
Does it? I'd think `force_autorelabel` would just work on the path given in `target:` (like it does today).

Reply:
Yes, did exactly like that. Originally i was thinking about such kind of configs without explicit `target`:
```
 - type: org.osbuild.selinux
      options:
        labels:
          mount://root/file: system_u:object_r:boot_t:s0
        force_autorelabel: 1
      mounts:
        - name: root
          source: disk
          target: /
```

Reply:
without explicit target it would need to do what it does today, which is create the `.autorelabel` file in the root of the tree.

with explicit target then it would just drop that file wherever the path of target is.